pkg/storage/snapshot/restore.go

```diff
@@ -571,6 +577,10 @@ func (t *vmRestoreTarget) reconcileSpec() (bool, error) {
 	}
 	t.vm = newVM
 
+	if err = t.claimInstancetypeControllerRevisionsOwnership(t.vm); err != nil {
+		return false, err
+	}
+
```

Comment:
Is there any trick we can do here to set the ownership on the controller revisions at creation time instead of afterwards?

For example, what would happen if we created a VM with a controller revision that didn't exist yet, and then posted the CR afterwards? Would that confuse the VM into thinking it needed to create a new CR because it couldn't find the revision temporarily?